app/callbacks.py

```diff
@@ -0,0 +1,195 @@
+import os
+import pickle
+import tempfile
+import uuid
+import dash
+import dash_bootstrap_components as dbc
+import dash_mantine_components as dmc
+import dash_uploader as du
+from dash import ALL
+from dash import MATCH
+from dash import Dash
+from dash import Input
+from dash import Output
+from dash import State
+from dash import callback_context as ctx
+from dash import clientside_callback
+from dash import dcc
+from dash import html
+
+
+dash._dash_renderer._set_react_version("18.2.0")
+
+
+dbc_css = "https://cdn.jsdelivr.net/gh/AnnMarieW/dash-bootstrap-templates/dbc.min.css"
+app = Dash(__name__, external_stylesheets=[dbc.themes.UNITED, dbc_css, dbc.icons.FONT_AWESOME])
+# Configure the upload folder
+TEMP_DIR = tempfile.mkdtemp()
+du.configure_upload(app, TEMP_DIR)
+
+clientside_callback(
+    """
+    (switchOn) => {
+    document.documentElement.setAttribute('data-bs-theme', switchOn ? 'light' : 'dark');
+    return window.dash_clientside.no_update
+    }
+    """,
+    Output("color-mode-switch", "id"),
+    Input("color-mode-switch", "value"),
+)
+
+
+@du.callback(
+    id="dash-uploader",
+    output=[Output("dash-uploader-output", "children"), Output("file-store", "data")],
+)
+def upload_data(status: du.UploadStatus):  # noqa: D103
+    if status.is_completed:
+        latest_file = status.latest_file
+        with open(status.latest_file, "rb") as f:
+            pickle.load(f)
+        return (
+            f"Successfully uploaded: {os.path.basename(latest_file)} [{round(status.uploaded_size_mb, 2)} MB]",
+            str(latest_file),
+        )
```

Comment:
`pickle.load` will fail if giving a fake file, e.g. change `file_mappings.tsv` to `file_mapping.pickle` and upload it, I got the following:
![image](https://github.com/user-attachments/assets/1e9b8b48-af5e-4500-8ba1-cbb5379a0d96)

To prevent this, it's better to add an error handling.

Reply:
Weird, if I do that by simply changing the .tsv file's name to somethig.pickle it keeps giving me: 
<img width="1134" alt="image" src="https://github.com/user-attachments/assets/340ff622-22e7-4440-a27e-2929b62016a4">
As if it detects that it's not a pickle file. 
Anyway, I added a try except block.